Setting OPAMFETCH leads to wrong arguments being passed to wget

Here's an example (using master), where setting `OPAMFETCH=wget` leads to the `curl` arguments `--insecure`, `--retry-delay`, etc. being passed to `wget:

```
$ OPAMFETCH=wget ./src/opam install -vv ocamlfind 
The following actions will be performed:
  ∗  install ocamlfind 1.5.5

=-=- Gathering sources =-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
[ocamlfind: from default] Command started
+ wget "--write-out" "%{http_code}\\n" "--insecure" "--retry" "3" "--retry-delay" "2" "-L" "-o" "/home/jeremy/.opam/repo/default/archives/ocamlfind.1.5.5+opam.tar.gz.part" "https://opam.ocaml.org/1.3/archives/ocamlfind.1.5.5+opam.tar.gz"
- wget: unrecognized option '--write-out'
- wget: unrecognized option '--insecure'
- wget: unrecognized option '--retry-delay'
- Usage: wget [OPTION]... [URL]...
- 
- Try `wget --help' for more options.
[ERROR] The sources of the following couldn't be obtained, aborting:
          - ocamlfind.1.5.5
        (This might be due to outdated metadata, in this case run 'opam update')
'./src/opam install -vv ocamlfind' failed.
```

--- a/src/repository/opamDownload.ml
+++ b/src/repository/opamDownload.ml
@@ -44,7 +44,7 @@ let download_args ~url ~out ~retry ?checksum ~compress =
   let cmd, _ = Lazy.force OpamRepositoryConfig.(!r.download_tool) in
   let cmd =
     match cmd with
-    | [CIdent "wget", _] -> cmd @ wget_args
+    | [(CIdent "wget"|CString "wget"), _] -> cmd @ wget_args
     | [_] -> cmd @ curl_args (* Assume curl if the command is a single arg *)
     | _ -> cmd
   in

